packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/LightCurveViewer/LightCurveViewer.js

```diff
@@ -40,37 +78,53 @@ class LightCurveViewer extends Component {
   }
 
   componentWillUnmount () {
-    this.d3interfaceLayer.on('.zoom', null)
+    this.d3interfaceLayer && this.d3interfaceLayer.on('.zoom', null)
   }
 
   clearChart () {
     this.d3dataLayer.selectAll('circle')
       .remove()
   }
 
+  /*
+  Updates the D3 chart to fit the size of the container, and adds/updates the
+  data points.
+  Called when new data (points) is received, and when chart is resized.
+   */
   drawChart (width, height, shouldAnimate = false) {
     if (!height || !width) {
       return false
     }
-
+    
+    /*
+    Limit zoom panning to x-direction (yMin=0, yMax=0), and don't allow panning
+    beyond the start (xMin=0) or end (xMax=width) of the light curve.
+    */
+    this.zoom.translateExtent([[0, 0], [width, 0]])
+
+    // Update x-y scales to fit current size of container
     this.xScale
       .domain(this.props.extent.x)
       .range([0, width])
-
     this.yScale
       .domain(this.props.extent.y)
-      .range([height, 0])
-
-    const points = this.d3dataLayer.selectAll('circle')
+      .range([height, 0])  //Note that this is reversed
+    
+    this.updateAxes()
+    
+    // Add the data points
+    const points = this.d3dataLayer.selectAll('.data-point')
       .data(this.props.points)
 
+    const t = this.getCurrentTransform()
     const setPointCoords = selection => selection
-      .attr('cx', d => this.xScale(d[0]))
+      // users can only zoom & pan in the x-direction
+      .attr('cx', d => t.rescaleX(this.xScale)(d[0]))
       .attr('cy', d => this.yScale(d[1]))
 
     if (shouldAnimate) {
       points.enter()
-        .append('circle')
+        .append('circle')  // Note: all circles are of class '.data-point'
```

Comment:
If that's the case, we should probably select on that class, otherwise there's no point having it at the moment as styling is done directly on the node. Also, that would probably be a good thing to optimise with a styled component.

Reply:
👌 Make sense. At the moment, the only place that select `<circle>`s is in
```
  clearChart () {
    this.d3dataLayer.selectAll('circle')
      .remove()
  }
``` 
and that's pretty straightforward for me to update. Gimme a sec.

As for using styled components, do you mean instead of...
```
points.enter()
  .append('circle')  // Note: all circles are of class '.data-point'
  .call(setPointStyle)
  ...
```
we do something like...?
```
const DataPoint = styled.circle`
  fill: ${props => (props.darkTheme) ? '#fff' : '#000'};
`

points.enter()
  .append(DataPoint)
```

I haven't tried combining `styled-components` with D3 yet, but that will be worth trying out in the optimisation phase.